tests/core/management/schedulers/test_base_scheduler.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, Iterable, Type, Union
+
+import pytest
+from immutables import Map
+from mock import Mock, create_autospec, patch
+
+from tickit.core.management.event_router import InverseWiring, Wiring
+from tickit.core.management.schedulers.base import BaseScheduler
+from tickit.core.state_interfaces.state_interface import StateConsumer, StateProducer
+from tickit.core.typedefs import (
+    Changes,
+    ComponentID,
+    Input,
+    Interrupt,
+    Output,
+    PortID,
+    SimTime,
+)
+
+
+class _TestBaseScheduler(BaseScheduler):
+    def __init__(
+        self,
+        wiring: Union[Wiring, InverseWiring],
+        state_consumer: Type[StateConsumer],
+        state_producer: Type[StateProducer],
+    ):
+        super().__init__(wiring, state_consumer, state_producer)
+        self.schedule_interrupt_called = False
+        self.schedule_interrupt_call_count = 0
+
+    async def schedule_interrupt(self, source: ComponentID) -> None:
+        self.schedule_interrupt_called = True
+        self.schedule_interrupt_call_count += 1
+        return None
```

Comment:
I did this to prevent calls to `schedule_interrupt` from throwing `NotImplementedError`. If there's a clean way to patch out those calls then that would probably be preferrable.

Reply:
Actually I did this to even be able to call the constuctor of `BaseScheduler`, which we cannot do otherwise since it has the abstract attribute `schedule_interrupt`.

Reply:
Could we do something along the lines of what's shown below?

```python
def mock_base_scheduler() -> BaseScheduler:
    with mock.patch(BaseScheduler, "schedule_interrupt") as mock_schedule_interrupt:
        return BaseScheduler()
```